4주차/16198/16198_java_잉.java

```diff
@@ -0,0 +1,43 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringTokenizer;
+
+public class BOJ16198 {
+    private static int maxEnergy = 0;
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st = new StringTokenizer(br.readLine());
+        int n = Integer.parseInt(st.nextToken());
+
+        List<Integer> list = new ArrayList<>();
+        st = new StringTokenizer(br.readLine());
+        for (int i = 0; i < n; i++) {
+            list.add(Integer.parseInt(st.nextToken()));
+        }
+
+        getMaxEnergy(list, 0, n);
+        System.out.println(maxEnergy);
+    }
+
+    private static void getMaxEnergy(List<Integer> list, int energy, int size) {
+        if (size == 2) {
+            maxEnergy = Math.max(maxEnergy, energy);
+            return;
+        }
+
+        for (int i = 1; i < list.size() - 1; i++) {
+            int newEnergy = energy + (list.get(i - 1) * list.get(i + 1));
+            getMaxEnergy(makeNextList(list, i), newEnergy, size - 1);
```

Comment:
```suggestion
            getMaxEnergy(list.remove(i), newEnergy, size - 1);
            numbers.add(i, num);
```
꼭 리스트를 복사하여 사용하지 않고 인덱스를 사용해서 넣고 뺄 수도 있을거 같아요🙂

Reply:
요런 이슈가 있었네요 잉의 방법이 더 좋네요👍👍👍👍👍
https://github.com/woowacourse-study/2022-lv3-algorithm-study/pull/138#discussion_r928378643